specs/fault-proof-system-v2/transaction-data-commitment.md

```diff
@@ -0,0 +1,146 @@
+# Transaction Data Commitment
+
+<!-- All glossary references in this file. -->
+
+[g-zk-fault-proof]: ../glossary.md#zk-fault-proof
+[g-l1-attr-deposit]: ../glossary.md#l1-attributes-deposited-transaction
+[g-user-deposited]: ../glossary.md#user-deposited-transaction
+[g-system-config]: ../glossary.md#system-configuration
+[g-l1-attr-predeploy]: ../glossary.md#l1-attributes-predeployed-contract
+[g-state]: ../glossary.md#state
+[g-avail-provider]: ../glossary.md#data-availability-provider
+[g-sequencer]: ../glossary.md#sequencer
+[g-batcher-transaction]: ../glossary.md#batcher-transaction
+[g-channel-frame]: ../glossary.md#channel-frame
+[g-channel]: ../glossary.md#channel
+[g-sequencer-batch]: ../glossary.md#sequencer-batch
+
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+**Table of Contents**
+
+- [Overview](#overview)
+- [Types of Transactions in Kroma L2 Block](#types-of-transactions-in-kroma-l2-block)
+- [Deposited Transactions](#deposited-transactions)
+  - [L1 Attributes Deposited Transactions](#l1-attributes-deposited-transactions)
+  - [User-Deposited Transactions](#user-deposited-transactions)
+- [L2 Transactions](#l2-transactions)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+## Overview
+
+This document outlines the process of on-chain verification of transaction data in the context of
+[ZK fault proof][g-zk-fault-proof]. It's essential for asserters or challengers to prove that their ZK fault proof was
+generated by correctly executing the relevant transactions of a target L2 block in the zkEVM.
+
+## Types of Transactions in Kroma L2 Block
+
+The Kroma L2 block comprises three types of transactions:
+
+- [L1 Attributes Deposited Transactions][g-l1-attr-deposit]
+- [User-Deposited Transactions][g-user-deposited]
+- L2 Transactions
+
+The sections below detail how commitments for these transactions are stored on-chain and utilized for ZK proof
+verification.
+
+## Deposited Transactions
+
+### L1 Attributes Deposited Transactions
+
+L1 attributes deposited transactions store information from the L1 reference block and
+[L2 system configuration][g-system-config] in the [L1Block][g-l1-attr-predeploy] contract. Automatically generated as
+the first transaction of every new L2 block, they change L2 [state][g-state] and hence must be included in ZK proof
+verification. However, they are not currently part of [batch data][g-sequencer-batch], making on-chain verification
+challenging. Future updates will address this with specific verification instructions.
+
+TBD
```

Comment:
Is this intentional?

Reply:
Yes, it is intended. This part is to be determined.